src/providers/hana/qgshanaprovider.cpp

```diff
@@ -1350,7 +1385,10 @@ QgsRectangle QgsHanaProvider::estimateExtent() const
       QString subQuery = buildQuery( QStringLiteral( "ST_EnvelopeAggr(%1) AS ext" ).arg( QgsHanaUtils::quotedIdentifier( mGeometryColumn ) ) );
       sql = QStringLiteral( "SELECT ext.ST_XMin(),ext.ST_YMin(),ext.ST_XMax(),ext.ST_YMax() FROM (%1)" ).arg( subQuery );
     }
+  }
 
+  try
+  {
     QgsHanaResultSetRef rsExtent = conn->executeQuery( sql );
```

Comment:
You could restructure the code as follows and get rid of one rsExtent->close() call.

```
    QgsHanaResultSetRef rsExtent = conn->executeQuery( sql );
    QgsRectangle ret;
    if ( rsExtent->next() )
    {
      QVariant val = rsExtent->getValue( 1 );
      if ( !val.isNull() )
      {
        ret.setXMinimum( val.toDouble() );
        ret.setYMinimum( rsExtent->getValue( 2 ).toDouble() );
        ret.setXMaximum( rsExtent->getValue( 3 ).toDouble() );
        ret.setYMaximum( rsExtent->getValue( 4 ).toDouble() );
      }
    }
    rsExtent->close();

   if (useEstimatedMetadata && ret.empty())
        return estimateExtent( false );

```

Reply:
Done.